airflow/www/views.py

```diff
@@ -3715,7 +3715,6 @@ def next_run_datasets(self, dag_id):
                     DatasetEvent,
                     and_(
                         DatasetEvent.dataset_id == DatasetModel.id,
-                        DatasetEvent.timestamp > DatasetDagRunQueue.created_at,
```

Comment:
@blag Does this look right to you? re: https://github.com/apache/airflow/pull/26356

Reply:
😬  It's been a long minute since I wrote this, but...

I believe when I wrote this the intent with the `lastUpdate` field was to only show last updates _since the last time the dag was queued/run_. But yeah, the `lastUpdate` label isn't descriptive enough for that.

Option 1: Personally, I would consider changing the name of what the `lastUpdate` field is rendered to, something like "Last update since last run" or something more wordy.

Option 2: But if you don't want to do that, and you want to display the last update for every dataset regardless of whether has already been "consumed" by a DagRun (eg: in either the DatasetDagRunQueue or actually scheduled into a DagRun), then yeah it makes sense to remove this filter. However, I would also remove the `and_` around it since then there would only be one filter condition in that join:

```python
                .join(
                    DatasetEvent,
                    DatasetEvent.dataset_id == DatasetModel.id,
                    isouter=True,
                )
```

If you go for option 2, I think you should be able to compare the existence and creation time of the DDRQ with the DatasetEvent timestamp to figure out whether or not the last update time has already triggered a DagRun or if it has partially satisfied the conditions of a future DagRun.

Hope this makes sense.

Reply:
> However, I would also remove the and_ around it since then there would only be one filter condition in that join:

Yes, you're right the `and` becomes unnecessary. 

I think there might be some confusion around DDRQ. My understanding is that when a `DatasetEvent` is created, a DDRQ record is created per consuming DAG. Then, once a DAG has an associated DDRQ record for each `Dataset` that it depends on, a dag_run is created and then all DDRQ records associated with that DAG are deleted. 

> If you go for option 2, I think you should be able to compare the existence and creation time of the DDRQ with the DatasetEvent timestamp to figure out whether or not the last update time has already triggered a DDRQ/DagRun or if it has partially satisfied the conditions of a future DagRun.

As I understand it, if there are DDRQ records for a DAG,  we can assume that there hasn't been a DagRun triggered since the last DatasetEvent (because we delete DDRQ records on the creation of a DagRun).

Reply:
@blag Does that make sense to you?

Reply:
I just dug through the code in more detail, and yes, your understanding seems to be correct. 😄